src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/VisitSchedulerService.kt

```diff
@@ -92,19 +94,35 @@ class VisitSchedulerService(
 
   fun createVisit(createVisitRequest: CreateVisitRequest): VisitDto {
     log.info("Creating visit for ${createVisitRequest.prisonerId}")
-    return VisitDto(
-      visitRepository.save(
-        Visit(
-          prisonId = createVisitRequest.prisonId,
-          prisonerId = createVisitRequest.prisonerId,
-          visitType = createVisitRequest.visitType,
-          visitRoom = createVisitRequest.visitRoom,
-          visitStart = createVisitRequest.startTimestamp,
-          visitEnd = createVisitRequest.endTimestamp,
-          status = VisitStatus.RESERVED
-        )
+    val visitEntity = visitRepository.saveAndFlush(
+      Visit(
+        prisonId = createVisitRequest.prisonId,
+        prisonerId = createVisitRequest.prisonerId,
+        visitType = createVisitRequest.visitType,
+        visitRoom = createVisitRequest.visitRoom,
+        visitStart = createVisitRequest.startTimestamp,
+        visitEnd = createVisitRequest.endTimestamp,
+        status = VisitStatus.RESERVED,
       )
     )
+
```

Comment:
very very minor style thing; I think this would read slightly better if you tagged on the the saveAndFlush something like 

`).apply { visit ->
 visit.vistors += contactList.map {
 visitorId -> VisitVistor(.....)
}
 
}`